### DISCUSSION TITLE
Visualizing a Color Swatch within a Standard L, a, b Color Palette

### DISCUSSION THREAD

**Discussion by shpurdikhub** [colour-science/colour#927] [Q&A]
Please, advise which of the Colour functions to use and how to visualize a color swatch defined by L, a, b coordinates, within the L, a, b color palette. I can also use the tricolor coordinates X, Y, Z or RGB values to define the swatch, but the reference palette must be in L, a, b format. An example image of the interface I am looking for is attached below and is adopted from Powerpoint. Thank you!
![Screenshot 2022-01-26 141020](https://user-images.githubusercontent.com/98533047/151369522-60c0d929-1692-4e47-a48e-f5da12d1b873.jpg)

**Comment by thomasmansencal** [comment]
Hi @shpurdikhub,

Is it fair to say that you are actually looking for a colour picker in CIE Lab?

Cheers,

Thomas

**Reply by shpurdikhub** [reply]
Yes, this is what I am trying to do, except I want to do it in Hunter Lab space if possible. If it is not possible, then I think I can make use of the CIE Lab space instead.

**Reply by thomasmansencal** [reply]
Generating the picture itself is not super hard, however the interactivity part requires quite some work. Is there a usage context? At any rate, I'm not sure any of us has time to implement something interactive atm.

**Reply by shpurdikhub** [reply]
Here is my usage context: I calculate a color of a mixture on a background and use this prediction to achieve a desired color by adjusting the component colors. This is important for the decorative printing on many products my company manufactures. The visualization part that I asked for your help with is used to understand how far the prediction is from the desired color. The L, a, b values by themselves do not provide a sensorial aspect of this evaluation, so the visualization part is necessary. Now to the interactive part: the task may be simpler than it seems if you take into account that I do not actually need to “pick” the color from the palette, only **show** this color in a right place on the palette using its L, a, b coordinates. Let me know whether this is doable given your time constraints.

**Reply by shpurdikhub** [reply]
Thomas - please, help me to figure out why I cannot change the size of the figure in colour.plotting.plot_single_colour_swatch command. I use the width and height parameters, but they only change the aspect ratio and not the physical size. If the aspect ratio is 1:1, the size does not change no matter which numbers I use for width and height.

**Reply by thomasmansencal** [reply]
The size can be set by setting `plt.rcParams["figure.figsize"]` directly or passing an existing `axes` at the right size. Typically we will set the default _colour_ style and then set a custom resolution if required:

```python
import colour
import matplotlib.pyplot as plt

colour.plotting.colour_style()
plt.rcParams.update({'figure.figsize': (19.20, 10.80)})
```
